src/battle_main.c

```diff
@@ -5174,7 +5174,11 @@ static void FreeResetData_ReturnToOvOrDoEvolutions(void)
                                   | BATTLE_TYPE_FRONTIER
                                   | BATTLE_TYPE_EREADER_TRAINER
                                   | BATTLE_TYPE_WALLY_TUTORIAL))
+#if !B_EVOLUTION_AFTER_WHITEOUT
             && (gBattleOutcome == B_OUTCOME_WON || gBattleOutcome == B_OUTCOME_CAUGHT))
+#else
+                                  )
```

Comment:
Not sure if this is good practice or not.

Reply:
Just chiming in, you might be able to do:
 `&& ((gBattleOutcome == B_OUTCOME_WON || gBattleOutcome == B_OUTCOME_CAUGHT) || B_EVOLUTION_AFTER_WHITEOUT)`
Not sure if that looks _that_ much better, but I think this would function the same.

Reply:
This can work, though I would put the config check before the outcome checks. Just a small optimization :P